### ISSUE TITLE
[Feature Request] Optional switch of "cutout" render type for "cutout_mipped" (optifine feature)

### ISSUE THREAD

**Issue by djmrFunnyMan** [FlashyReese/sodium-extra-fabric#427]
This feature was recently rejected from Sodium: https://github.com/CaffeineMC/sodium/issues/2846

It's been a part of optifine since a while ago.

TLDR: Vanilla disables mipmapping for a lot of cutout blocks, which causes atrocious aliasing.
![2024-10-18_15 54 38](https://github.com/user-attachments/assets/85cf4e94-c397-49bc-ac55-c438f3e06748)

Optifine remedies this and automatically switches blocks using "cutout" render type for "cutout_mipped".
 
Since there is no maintained standalone mod to add this behaviour to sodium I think sodium extra should add it, considering the description says "Sodium Extra aims to bring most of OptiFine's eye-candy options to Sodium" (and fixing this eyesore would definitely count as eye-candy)

There is a handful of exceptions that do need to use "cutout" render type. That being glass, redstone wire and chains, because these blocks melt away into nothing with mips. Those should be forced to keep using the "cutout" render type, either via hardcoding those block ID's (what optifine does) or putting them into a config file. The latter would be preferred, as it'd allow users to exclude any modded blocks which might run into issues when mipmapping is forced on.

Personally I think this should be on by default, (like in optifine) but that's up to you.
